botorch/models/model.py

```diff
@@ -244,6 +245,10 @@ def train(self, mode: bool = True) -> Model:
             self._set_transformed_inputs()
         return super().train(mode=mode)
 
+    @property
+    def dtypes_of_buffers(self) -> Set[torch.dtype]:
+        return {t.dtype for t in self._buffers.values() if t is not None}
```

Comment:
Why not use the `.buffers()` iterator here? 
```suggestion
        return {t.dtype for t in self.buffers() if t is not None}
```